### PR TITLE
prov/verbs: Fix invalid usage of IBV_SEND_INLINE

### PR DESCRIPTION
**Pull request by a-ilango**
- Flag IBV_SEND_INLINE is valid only for IBV_WR_SEND and
  IBV_WR_RDMA_WRITE opcodes. Don't set it for other opcodes.
- Fix MR access permissions corresponding to FI_REMOTE_WRITE
   to allow atomic ops.

Signed-off-by: Arun C Ilango <arun.ilango@intel.com>

### CODE REVIEW

**Comment by a-ilango** [comment]
@shefty Can you please review?

File: prov/verbs/src/fi_verbs.c [hunk 1]
```diff
@@ -2085,9 +2097,16 @@ static int
...
 {
+	struct fi_ibv_msg_ep *_ep = container_of(ep, struct fi_ibv_msg_ep, ep_fid);
+
 	switch (op) {
 	case FI_ATOMIC_READ:
 	case FI_SUM:
+		if (_ep->info->tx_attr->op_flags & FI_INJECT) {
+			FI_WARN(&fi_ibv_prov, FI_LOG_EP_DATA,"FI_INJECT not "
+				"supported for fi_fetch_atomic with FI_SUM op\n");
+			return -FI_EINVAL;
```
Comment (shefty) [inline prov/verbs/src/fi_verbs.c]: Note that this is also handling the case FI_ATOMIC_READ

**Comment by shefty** [comment]
Just one comment, otherwise looks okay.

File: prov/verbs/src/fi_verbs.c (hunk 1 above)
Comment (a-ilango) [inline prov/verbs/src/fi_verbs.c]: Thanks! Will fix it.

**Comment by a-ilango** [comment]
Updated the PR with suggested changes.

**Comment by shefty** [comment]
Looks okay to me.

**Pull request merged**